File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int tar) {
        for(int i=0;i<nums.size();++i){
            int low = i+1,high = nums.size()-1;
            while(low<=high){
                int mid = low + (high-low)/2;
                if (nums[mid]+nums[i]==tar) return {i+1,mid+1};
                else if (nums[mid] + nums[i] > tar) high = mid-1;
                else if (nums[mid] + nums[i]< tar) low = mid+1;
                // cout<< i<<mid<< low<<high<<endl;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pnums = [ [nums[i],i] for i in range(len(nums))]
        pnums.sort()
        low, high = 0, len(nums)-1
        while low < high:
            if pnums[low][0] + pnums[high][0] == target:
                return sorted([pnums[low][1],pnums[high][1]])
            elif pnums[low][0] + pnums[high][0] > target:
                high -=1
            else :
                low +=1
        return []